Sources/WalletConnectSign/Types/ProtocolMethods/SessionProposeProtocolMethod.swift

```diff
@@ -2,8 +2,45 @@ import Foundation
 
 struct SessionProposeProtocolMethod: ProtocolMethod {
     let method: String = "wc_sessionPropose"
+    
+    enum Tag: Int {
+        case sessionPropose = 1100
+        case sessionProposeResponseApprove = 1101
+        case sessionProposeResponseReject = 1120
+        case sessionProposeResponseAutoReject = 1121
+    }
 
-    let requestConfig = RelayConfig(tag: 1100, prompt: true, ttl: 300)
+    let requestConfig: RelayConfig
 
-    let responseConfig = RelayConfig(tag: 1101, prompt: false, ttl: 300)
+    let responseConfig: RelayConfig
+   
+    static let defaultTtl: TimeInterval = 300
+    
+    private init(
+        ttl: TimeInterval,
+        responseTag: Tag
+    ) {
+        self.requestConfig = RelayConfig(
+            tag: Tag.sessionPropose.rawValue,
+            prompt: true,
+            ttl: Int(ttl)
+        )
+        self.responseConfig = RelayConfig(
+            tag: responseTag.rawValue,
+            prompt: false,
+            ttl: Int(ttl)
+        )
+    }
+    
+    static func responseApprove(ttl: TimeInterval = Self.defaultTtl) -> Self {
+        Self(ttl: ttl, responseTag: .sessionProposeResponseApprove)
+    }
+    
+    static func responseReject(ttl: TimeInterval = Self.defaultTtl) -> Self {
```

Comment:
```suggestion
    static func responseUserReject(ttl: TimeInterval = Self.defaultTtl) -> Self {
```
would it be better to be more specific?

Reply:
I'm using `responseReject` for user rejected and `responseAutoReject` for protocol level auto rejections. I agree that this naming could be clearer but I also wanted to keep things aligned with [`Kotlin`](https://github.com/WalletConnect/WalletConnectKotlinV2/pull/1394/files#diff-c384e6db5880e14f35640a2d0a5ca3929714398868a15de74ce79c87cc8d4c85R14)